gobblin-cluster/src/main/java/org/apache/gobblin/cluster/HelixRetriggeringJobCallable.java

```diff
@@ -126,8 +126,17 @@ public HelixRetriggeringJobCallable(
     this.planningJobLauncherMetrics = planningJobLauncherMetrics;
     this.helixMetrics = helixMetrics;
     this.appWorkDir = appWorkDir;
-    this.jobHelixManager = jobHelixManager;
-    this.taskDriverHelixManager = taskDriverHelixManager;
+
+    // make a copy of helix managers, so that each job can keep using it without worrying about disconnect from the GobblinClusterManager where they are created
+    this.jobHelixManager = GobblinHelixMultiManager.buildHelixManager(ConfigUtils.propertiesToConfig(sysProps),
+        sysProps.getProperty(GobblinClusterConfigurationKeys.ZK_CONNECTION_STRING_KEY),
+        GobblinClusterConfigurationKeys.HELIX_CLUSTER_NAME_KEY, jobHelixManager.getInstanceType());
+
+    this.taskDriverHelixManager = taskDriverHelixManager.map(
+        helixManager -> GobblinHelixMultiManager.buildHelixManager(ConfigUtils.propertiesToConfig(sysProps),
+            sysProps.getProperty(GobblinClusterConfigurationKeys.ZK_CONNECTION_STRING_KEY),
+            GobblinClusterConfigurationKeys.TASK_DRIVER_CLUSTER_NAME_KEY, helixManager.getInstanceType()));
+
```

Comment:
when I saw this above, I was wondering whether it shouldn't be abstracted into a utility... now seeing it again (twice!), I really think so.  any impediment to doing so?

Reply:
The thing I considered is that I was trying to copy a HelixManager, not a ZKHelixManager. That was all the reasoing behind using the utility method.

Reply:
yeah, I suppose `GobblinHelixMultiManager.buildHelixManager` is already a utility, probably excessive to abstract too much beyond.  what I noticed is that (before calling) you're pulling a property out of the same `Config` you provide to the call.  usage shows `ZK_CONNECTION_STRING_KEY` to be pretty widely referenced... what about a 3-param overload:
```
protected static HelixManager buildHelixManager(Config config, String clusterName, InstanceType type) {
  return buildHelixManager(config, config.getProperty(GobblinClusterConfigurationKeys.ZK_CONNECTION_STRING_KEY), clusterName, type);
}
```
?
  

is

Reply:
I agree, the existing code did not look good to me as well. will change